src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/SubscriptionLevelDeploymentTests.ps1

```diff
@@ -0,0 +1,51 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Tests deployment template validation.
+#>
+function Test-DeploymentEndToEnd
+{
```

Comment:
I tried to test them but they are not easily testable. Test-AzureRmDeployment generates a random deploymentName every time. so the captured traffic doesn't work because every time it looks for a different deployment name. Stop-azureRmDeployment requires the deployment to be active to test it. This is difficult to simulate in scenario tests because New-AzureRmDeployment always wait the deployment to finish. 

I manually tested all of them, though.

Reply:
To avoid waiting for the ```New-AzureRmDeployment``` cmdlet to complete - run it with the ```-AsJob``` switch.

Reply:
@Tiano2017 
Please remind me what is the reason not to test the ```Save-AzureRmDeploymentTemplate``` cmdlet?